### PR TITLE
*: support Alertmanager API v2

### PR DESCRIPTION
**Pull request by simonpasquier**
* [X] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

Added a new parameter to the alerting configuration similar to what we have in the Prometheus configuration.

## Verification

End-to-end tests modified to verify both v1 and v2 endpoints.

### CODE REVIEW

File: pkg/alert/alert.go [hunk 1]
```diff
@@ -354,7 +405,7 @@ func (s *Sender) Send(ctx context.Context, alerts []*Alert) {
 	}
 
 	s.dropped.Add(float64(len(alerts)))
-	level.Warn(s.logger).Log("msg", "failed to send alerts to all alertmanagers", "alerts", string(b))
+	level.Warn(s.logger).Log("msg", "failed to send alerts to all alertmanagers")
```
Comment (GiedriusS) [inline pkg/alert/alert.go]: A very small thing but maybe there is still some worth in logging how many alerts we have failed to send?

File: docs/components/rule.md [hunk 2]
```diff
@@ -416,6 +416,7 @@ alertmanagers:
   scheme: http
   path_prefix: ""
   timeout: 10s
+  api_version: v1
```
Comment (GiedriusS) [inline docs/components/rule.md]: Maybe we could add one sentence on what possible values are supported i.e. `v1` and `v2`?

File: CHANGELOG.md [hunk 3]
```diff
@@ -50,6 +50,7 @@ Compactor now properly handles partial block uploads for all operation like rete
 - [#1904](https://github.com/thanos-io/thanos/pull/1904) Add a skip-chunks option in Store Series API to improve the response time of `/api/v1/series` endpoint.
 - [#1910](https://github.com/thanos-io/thanos/pull/1910) Query: `/api/v1/labels` now understands `POST` - useful for sending bigger requests
 - [#1939](https://github.com/thanos-io/thanos/pull/1939) Ruler: Add TLS and authentication support for query endpoints with the `--query.config` and `--query.config-file` CLI flags. See [documentation](docs/components/rule.md/#configuration) for further information.
+- [#1982](https://github.com/thanos-io/thanos/pull/1982) Ruler: Add support for Alertmanager v2 API endpoints.
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply CHANGELOG.md]: done :+1:

File: docs/components/rule.md (hunk 2 above)
Comment (simonpasquier) [reply docs/components/rule.md]: added below as I don't know how we can add it directly here.

File: pkg/alert/alert.go (hunk 1 above)
Comment (simonpasquier) [reply pkg/alert/alert.go]: done

File: docs/components/rule.md (hunk 2 above)
Comment (GiedriusS) [reply docs/components/rule.md]: Yes, that's what I mean - thank you 👍

**Comment by simonpasquier** [comment]
Thanks!

**Pull request merged**